config/auth_u2m.go

```diff
@@ -0,0 +1,146 @@
+package config
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/databricks/databricks-sdk-go/config/credentials"
+	"github.com/databricks/databricks-sdk-go/credentials/u2m"
+	"github.com/databricks/databricks-sdk-go/credentials/u2m/cache"
+	"github.com/databricks/databricks-sdk-go/logger"
+	"golang.org/x/oauth2"
+)
+
+// u2mCredentials is a credentials strategy that uses the U2M OAuth flow to
+// authenticate with Databricks. It loads a token from the token cache for the
+// given workspace or account, refreshing it if needed. If the user has not
+// authenticated with OAuth U2M, it falls back to the next credentials strategy.
+// If they have but their access and refresh tokens are both invalid, it returns
+// a special error message that instructs the user how to reauthenticate.
+type u2mCredentials struct {
+	// auth is the persistent auth object that manages the token cache.
+	auth *u2m.PersistentAuth
```

Comment:
Should this just be a token source?

```suggestion
	ts oauth2.TokenSource
```

Reply:
Absolutely right, in the context of the SDK, we never use the challenge component of PersistentAuth, so we only depend on the TokenSource interface.